delivery.yaml

```diff
@@ -1,9 +1,142 @@
+        cmd: |
+          echo "Determining next version to publish:"
+          current_release=$(git describe --tags origin/master)
+          if [[ $(cut -d '-' -f1 <<< $current_release)  =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            version="$(cut -d '-' -f1 <<< $current_release)"
+            echo "Previous release: $version"
+            minorVersion="$(cut -d '.' -f2 <<< $version)"
+            majorVersion="$(cut -d '.' -f1 <<< $version)"
+            version="$majorVersion.$(($minorVersion + 1)).0"
+          else
+            echo "Issue with current tag: $current_release"
+            return -1
+          fi
+          sbt "set version in ThisBuild := \"$(cut -c2- <<< $version)\"" docker:publish -Ddocker.repo=registry-write.opensource.zalan.do/machina
+          echo "Published a new release version: $version"
+          echo "Tagging new version: $version"
+          git gh-tag $version
+
+  - id: release-major-version
+    type: script
+    when:
+      event: push
+      branch: master
+      triggered_by_dependency: null
+    requires_human_approval: true
+    vm_config:
+      type: linux
+      image: cdp-runtime/jdk8-sbt
+      size: large
+    commands:
+      - desc: Build and push docker image
+        cmd: |
+          echo "Determining next version to publish:"
+          current_release=$(git describe --tags origin/master)
+          if [[ $(cut -d '-' -f1 <<< $current_release)  =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            version="$(cut -d '-' -f1 <<< $current_release)"
+            echo "Previous release: $version"
+            buildVersion="$(cut -d '.' -f3 <<< $version)"
+            minorVersion="$(cut -d '.' -f2 <<< $version)"
+            majorVersion="$(cut -d '.' -f1 <<< ${version//v})"
+            version="v$((majorVersion + 1)).0.0"
+          else
+            echo "Issue with current tag: $current_release"
+            return -1
+          fi
+          sbt "set version in ThisBuild := \"$(cut -c2- <<< $version)\"" docker:publish -Ddocker.repo=registry-write.opensource.zalan.do/machina
+          echo "Published a new release version: $version"
+          echo "Tagging new version: $version"
+          git gh-tag $version
+
+  - id: release-updated-base-image
+    type: script
+    when:
+      event: push
+      branch: master
+      triggered_by_dependency: base-image
+    vm_config:
+      type: linux
+      image: cdp-runtime/jdk8-sbt
+      size: large
     commands:
       - desc: Build and push docker image
         cmd: |
-          sbt compile test docker:publish -Ddocker.repo=registry-write.opensource.zalan.do/machina
+          echo "Determining next version to publish:"
+          current_release=$(git describe --tags origin/master)
+          if [[ $(cut -d '-' -f1 <<< $current_release)  =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            version="$(cut -d '-' -f1 <<< $current_release)"
+            echo "Previous release: $version"
+            version="$version-$(date +'%Y%m%d')"
```

Comment:
This is specifically intended to re-release the same Git commit with new Docker base images so commit hash would cause collisions

Reply:
This will stops us to publish the same git commit twice in a single day, in case a base image is released multiple times in a day.
Shall we still include git hash so it's easier to link re-released image with the commit?

Reply:
In the case where we have multiple base image releases in a single day:
 - if the second attempt to publish fails and the second base image change is an unimportant update, then there's no issue with this step failing.
 - If the second attempt to publish fails and the second base image is important, we can go down the 'emergency' route of publishing the Docker image ourselves and marking it as production ready using Zalando Pierone APIs.
 - If we introduced Git hashes, we would require a whitespace change and approval from someone to trigger a valid publish. That requires two people (two approvals in Zappr), whereas publishing manually and marking as production ready only takes a single person.

We _could_ introduce CDP build versions instead of Git hashes which would completely avoid any issues around publishing multiple times in a single day, but I don't really want to leak more Zalando internal concepts into an opensource project.

Reply:
As I have your approval now, I'm going to proceed with this PR as is. If base image updates becomes an issue, we can then adjust this as needed.